### PR TITLE
ref(chat): on native, show messages as grouped by sender

### PR DESCRIPTION
**Pull request by virtuacoplenny**
grouped_chat

### CODE REVIEW

**Comment by virtuacoplenny** [comment]
Built on top of https://github.com/jitsi/jitsi-meet/pull/4173.

**Comment by virtuacoplenny** [comment]
Force pushed to resolve conflicts with master after chat css changes were merged in.

File: react/features/chat/components/web/Chat.js [hunk 1]
```diff
@@ -88,6 +87,37 @@ class Chat extends AbstractChat<Props> {
...
     }
 
+    /**
+     * Iterates over all the messages and creates nested arrays which hold
+     * consecutive messages sent be the same participant.
+     *
+     * @private
+     * @returns {Array<Array<Object>>}
+     */
+    _getMessagesGroupedBySender() {
```
Comment (zbettenbuk) [inline react/features/chat/components/web/Chat.js]: Why do you have it twice? A leftover from a refactor?

File: react/features/chat/components/web/Chat.js [hunk 2]
```diff
@@ -96,17 +126,11 @@ class Chat extends AbstractChat<Props> {
...
      */
     _renderChat() {
-        const messages = this.props._messages.map(this._renderMessage);
-
-        messages.push(<div
-            key = 'end-marker'
-            ref = { this._setMessageListEndRef } />);
-
         return (
             <>
```
Comment (zbettenbuk) [inline react/features/chat/components/web/Chat.js]: What are these?

File: react/features/chat/components/web/MessageContainer.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    from '../AbstractMessageContainer';
+
+import ChatMessageGroup from './ChatMessageGroup';
+
+/**
+ * Displays all received chat messages, grouped by sender.
+ *
+ * @extends AbstractMessageContainer
+ */
+export default class MessageContainer extends AbstractMessageContainer<Props> {
```
Comment (zbettenbuk) [inline react/features/chat/components/web/MessageContainer.js]: Maybe it's better not to implement a parametrized class if you don't need to extend or override the props

File: react/features/chat/components/web/Chat.js (hunk 2 above)
Comment (virtuacoplenny) [reply react/features/chat/components/web/Chat.js]: React fragments to avoid an extra div.

File: react/features/chat/components/web/Chat.js (hunk 1 above)
Comment (virtuacoplenny) [reply react/features/chat/components/web/Chat.js]: Looks like leftover. Initially the web implementation had the grouping done in Chat.js. But to make web more like mobile this logic got moved over to MessageContainer. I'm pretty sure this instance of _getMessagesGroupedBySender doesn't get called.

**Comment by virtuacoplenny** [comment]
Force pushed to rebase.

**Comment by virtuacoplenny** [comment]
Force pushed to rebase on fix for long message wrapping (https://github.com/jitsi/jitsi-meet/pull/4197) and to redo that same fix on top of these new changes.

**Pull request merged**